src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/food/Juice.java

```diff
@@ -65,39 +61,12 @@ public ItemConsumptionHandler getItemHandler() {
              * Minecraft has been broken when it comes to Saturation potions for a long time
              */
             for (PotionEffect effect : effects) {
-                if (effect.getType().equals(PotionEffectType.SATURATION)) {
-                    p.addPotionEffect(effect);
-                    break;
-                }
+                p.addPotionEffect(effect);
+                break;
             }
 
-            removeGlassBottle(p, item);
+            e.setItem(null);
```

Comment:
This will remove all 10 if they're stacked, why change this?

Reply:
I guess if there was something in the server that allowed stacking juices, this would break it, but the original removeGlassBottle method wouldn't work properly in that case either if the player had glass bottles in their inventory already or if the player had two juices in one stack.

Reply:
The removeGlassBottle is dumb I agree; we should probably just cancel the event, apply the effects, remove one from e.getItem() since we already know it's the correct hand